### PR TITLE
Add a pref to display or not original image in attachment preview

### LINKED ISSUE #3391: Create a pref to disable showing original image in attachment preview

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #3391

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/Attachments/Preview.tsx [hunk 1]
```diff
@@ -50,7 +52,7 @@ function Thumbnail({
           ? attachment.title
           : thumbnail.alt
       }
-      className={`
+      className={`${className}
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Attachments/Preview.tsx:55]: can you put this on a new line?

File: specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx [hunk 2]
```diff
@@ -46,6 +47,12 @@ export function AttachmentViewer({
     false
   );
 
+  const [displayAttachment] = userPreferences.use(
+    'attachments',
+    'behavior',
+    'display'
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx:53]: rename to `displayOriginal` or something more helpful like that

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]
minor changes

File: specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx (hunk 2 above)
Comment (maxxxxxdlp) [reply specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx:53]: I meant replace `'display'` user pref name with `'displayOriginal'`

and the variable name should stay `displayOriginal` like it is now

**Review by maxxxxxdlp — APPROVED** [review]

**Comment by chanulee1** [comment]
How do I test this?

**Comment by CarolineDenis** [comment]
1) Go in data entry form and add an attachment, the image should be displayed full size 
2) Go in preferences and unchecked the display original attachment pref 
3) Save 
4) Go in data entry form and add an attachment, the image should be displayed as a thumbnail

File: specifyweb/frontend/js_src/lib/localization/preferences.ts [hunk 3]
```diff
@@ -1413,4 +1413,7 @@ export const preferencesText = createDictionary({
     'fr-fr': 'Utiliser des infobulles modernes',
     'uk-ua': 'Використовувати сучасні підказки',
   },
+  displayOriginalAttachment: {
+    'en-us': 'Display original in preview',
```
Comment (grantfitzsimmons) [inline specifyweb/frontend/js_src/lib/localization/preferences.ts:1417]: Can this be instead a clearer preference:

image

Something that looks and works like the one above. This communicates better what this does to the user.

**Review by grantfitzsimmons — COMMENTED** [review]
It works as expected, though I think it is unclear what the preference does to a user upon first glance. Can we change the preference name and toggle? I can approve at that point

File: specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx [hunk 4]
```diff
@@ -93,49 +100,61 @@ export function AttachmentViewer({
...
+                alt={title}
+                className="h-full w-full object-scale-down"
+                src={thumbnail?.src}
+              />
+            </object>
+          )}
+        </div>
+      ) : (
+        <div className="flex items-center">
+          <span className="-ml-2 flex-1" />
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx:149]: span should not be needed here. justify-center instead?

File: specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx [hunk 5]
```diff
@@ -93,49 +100,59 @@ export function AttachmentViewer({
...
+              <img
+                alt={title}
+                className="h-full w-full object-scale-down"
+                src={thumbnail?.src}
+              />
+            </object>
+          )}
+        </div>
+      ) : (
+        <div className="flex items-center justify-center">
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx:148]: can you refactor this file to use the same `<div>` for both cases?

something like this:
```tsx
<div className="flex min-h-[30vw] w-full min-w-[30vh] flex-1 items-center">
      {displayOriginal === 'full' ?
         originalUrl === undefined ? (
            loadingGif
          ) : type === 'image' ? (
            <img alt={title} src={originalUrl} />
          )
         // ...
        : <Thumbnail
            attachment={serializeResource(attachment)}
            className="!border-none"
            thumbnail={thumbnail}
          />
</div>
```

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]

File: specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx (hunk 5 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/Attachments/Viewer.tsx]: Yes because I forgot to push it :D

**Review by maxxxxxdlp — APPROVED** [review]

**Comment by bronwyncombs** [comment]
An error was found in testing this. It may be an issue with the attachment servers, though I don't know much about that. 
I will say that this error was consistent across the issue branch #3391, v7.8.11, and v7.8.12prerelease, and in the databases for KUFish, KUMammals, and Cornell Fish. 

**To Recreate** 
1. Open any data entry form
2. Attach an image
3. See error

**Screenshot**
Screen Shot 2023-06-08 at 10 03 11 AM

**Comment by CarolineDenis** [comment]
@bronwyncombs this PR can be tested again

**Review by bronwyncombs — APPROVED** [review]
Working as expected! Changing attachment preferences from full-resolution to thumbnail updates existing and newly uploaded images in forms. 

https://github.com/specify/specify7/assets/135047322/2e4b53dd-4c32-4f27-9da6-6f47852fc3d6

**Pull request merged**